server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -310,7 +319,13 @@ public void startThreadResourceTracking(long threadId, ResourceStatsType statsTy
         }
         threadResourceInfoList.add(new ThreadResourceInfo(threadId, statsType, resourceUsageMetrics));
         incrementResourceTrackingThreads();
-        resourceTrackingListeners.forEach(listener -> listener.onTaskExecutionStartedOnThread(this, threadId));
+        resourceTrackingListeners.forEach(listener -> {
+            try {
+                listener.onTaskExecutionStartedOnThread(this, threadId);
+            } catch (Exception e) {
+                logger.warn("failure in listener during handling of onTaskExecutionStartedOnThread", e);
+            }
+        });
```

Comment:
Why silently catch a failure? Can we instead run everyone run, record failed ones and rethrow at the end
You can use ` ActionListener.onResponse(listeners, result);`
```
static <Response> void onResponse(Iterable<ActionListener<Response>> listeners, Response response) {
        List<Exception> exceptionList = new ArrayList<>();
        for (ActionListener<Response> listener : listeners) {
            try {
                listener.onResponse(response);
            } catch (Exception ex) {
                try {
                    listener.onFailure(ex);
                } catch (Exception ex1) {
                    exceptionList.add(ex1);
                }
            }
        }
        ExceptionsHelper.maybeThrowRuntimeAndSuppress(exceptionList);
    }
```

Reply:
Done!